Krav Maga Code Update

PKPenguin321 opened the pull request:
## THIS PR NO LONGER GIVES SEC KRAV MAGA

**See https://github.com/tgstation/-tg-station/pull/16843 for more info on Krav Maga**
- Gives sec an advantage over the crew that isn't some lameo baton
- Can be stolen since it's gloves and not an implant, opens it up to traitorous uses
- Krav is pretty much the weakest martial art we have so it's not OMG SLEEPIGN CARP FOR SEC REEE
- Traitor item version coming in a different PR (atomized features)

^ignore all of that stuff this doesnt give sec krav any more

Actual code changes in this PR:
- Commits sprites that I accidentally left out to actions.dmi
- Reduces leg sweep stun to Weaken(2)
- Creates and gives to sec a subtype of krav maga gloves that are very obvious, repaths the gloves to be children of black gloves
- Makes the implant toggle your krav maga on and off instead of just being one-use
- Adds myself to admins.txt
- Adds sprites for Krav Maga gloves, both ongrounds and onmobs

![krav maga gloves](https://cloud.githubusercontent.com/assets/8345184/14593435/3f1baf3e-04df-11e6-90e5-29651ad86ea9.png)

PKPenguin321 committed:
Gives sec officers krav maga

By equipping it on them roundstart

Core0verload commented:
Aaaand in the downvotes hell you go.

PKPenguin321 commented:
dank feedback bro

Supermichael777 commented:
AND SO THE PAIN TRAIN OF ANTI SEC GRUDGECODE GETS MORE FUEL.

steelrifle all over again

PKPenguin321 commented:
ehhhh not really, steelrifles were basically instakill ggnore le loyalty implant :^^^)

Supermichael777 commented:
steel-rifles never got round-start sec pins because even steel saw how shitty that would be.
the armory remapping just made everyone butt-hurt.

octareenroon91 commented:
Are they still heatproof?

Zilenan91 commented:
What are the Krav Maga moves again? Also yeah the gloves need to be heatproof if you can make them.

PKPenguin321 commented:
question: should i give this to roundstart nuke ops too

Iamgoofball commented:
get in coderbus

Core0verload commented:
Why can't sec players just git gud? Seriously, the amount of powercreep starts to worry me.

PKPenguin321 commented:
i'm not a sec player i'm like borderline antisec, i just think this is hella cool

Core0verload commented:
I remember a huge argument against stungloves was "OP UNDISARMABLE STUN NERF PLZ". And this resulted in stungloves being removed. And this PR is stungloves all over again, except now they are sec roundstarts.

What can possibly go wrong?

Zilenan91 commented:
This isn't stungloves though

PKPenguin321 commented:
stungloves could be carried by literally anybody with pretty much no tell, plus they had a full stunbaton level stun

i will make some tweaks to krav maga and the gloves to reduce the sekrit stun potential though

PKPenguin321 committed:
Merge branch 'patch-2' of https://github.com/PKPenguin321/-tg-station into patch-2

oh god what am i doing i cannot into git
MAKE THE COMMIT

PKPenguin321 committed:
krav update
1. actually adds icons that i forgot to commit to actions.dmi
2. makes leg sweep only do weaken(2) in krav_maga.dm, also adds a /sec subtype of krav maga gloves with a more obvious name
3. makes sec start with the /sec gloves in security.dm
4. makes the krav maga implant a toggle instead of giving you krav maga and deleting itself in implant_krav_maga.dm
5. adds my self to admins.txt

Core0verload commented:
You, four days ago:
>balance tag
on an admin only feature

So admin-only, much balance.

Bawhoppen commented:
pkp I will only allow this if the gloves are a good sprite
pic

PKPenguin321 commented:
gloves are normal sprite give better sprite i add it

@Core0verload at the time it most definitely was admin only

Iamgoofball commented:
@Bawhoppen they're just default gloves right now because i didn't have any

@PKPenguin321 if you want you could use the Hallway Brawler glove sprites

PKPenguin321 commented:
that is smart

KorPhaeron commented:
I can't decide if security running around punching everyone to death is funny or really stupid

Zilenan91 commented:
They have this on Hippie because of the tiding since they're on the hub. They might not need it here but it does seem pretty cool.

Fox-McCloud commented:
>should i give this to roundstart nuke ops too

Absolutely

Bawhoppen commented:
don't give people unneeded or unwanted things, ops do not need karate gloves and it does not fit their theme

PKPenguin321 committed:
makes sec gloves way more obvious
1. makes krav maga gloves pathed under black gloves for fire resistance
2. adds sprites for krav maga gloves to hands.dmi and gloves.dmi
3. sets up a var to prevent krav gloves from being cut into fingerless gloves in color.dm
4. makes sec use the new path for krav gloves in security.dm

PKPenguin321 commented:
@Bawhoppen krav maga is less "SUPER MEGA KUNG FU ACTION FIGHT MOVIE" and more "basic police combat training" (i *may* have overhyped the kung fu aspect in the irc)

optimumtact commented:
![It's time to stop](http://i0.kym-cdn.com/photos/images/original/001/060/689/927.jpg)

KorPhaeron commented:
@Cheridan still hasn't recieved 200 dollars, so I  believe stun gloves of any sort are still banned

PKPenguin321 commented:
Don't meme me

PKPenguin321 committed:
fixes conflicts

PKPenguin321 commented:
Conflcits should be fixed

optimumtact commented:
![stop this madman](https://file.house/cjH9.jpg)

KazeEspada commented on code/game/objects/items/weapons/implants/implant_krav_maga.dm:
```diff
@@ -19,8 +19,10 @@
+	if(istype(H.martial_art, /datum/martial_art/krav_maga))
+		style.remove(H)
+	else
+		style.teach(H,1)
```
Remove Krav Maga if they have it?

PKPenguin321 replied:
yeah, so you can toggle it

KazeEspada commented on config/admins.txt:
```diff
@@ -112,3 +112,4 @@ Lumipharon = Game Master
 bgobandit = Game Master
 coiax = Game Master
 RandomMarine = Game Master
+PKPenguin321 = Game Master
```
Stealth nerf

PKPenguin321 replied:
stop memeing me reeeeeeeeeeeeeee

Nanotransistor commented:
Lets give the guys that already have armor, ranged AoE stuns, and disabler smgs a weapon that cannot be disarmed and doesnt need to be equiped to be used (sec will always have these gloves on).


This kind of stuff would only be ok if traitor sec was enabled to balance it out.

PKPenguin321 commented:
ty for real feedback
i am considering just making it a traitor/nuke op only item now but i'd still like player input on this PR
if anybody could run a poll ingame (cough @KorPhaeron) that would be super cool of you

PKPenguin321 commented:
yo yo yo check out the [forum thread/poll](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=6466) right now and give a vote or something

right now it's looking like this will be given to only the warden instead of the entire sec team so dont get your jimmies too rustled

Zilenan91 commented:
It's tied with letting all sec officers have it. I see zero reason to not add it to everyone, it's not that powerful.

Fox-McCloud commented:
Should the detective and lawyer? (lol) have this?

Zilenan91 commented:
Lawyer isn't sec, and the detective shouldn't have it either since he shouldn't be security in the first place.

PKPenguin321 commented:
nah detective gets a pew pew gun he doesnt need krav on top of it, plus he's only like borderline sec anyways
lawyers are definitely no but it's a funny thought

demonfiren commented:
Lawyers only need neuralysers, nothing else.

Fox-McCloud commented:
Yeah, I know, it was mostly a joke =p

Fox-McCloud commented:
Still think nukies should have a box of these.

Also, should there be a box of these in the armory for people who change jobs to sec? That's largely what the tasers in the armory are for.

Bawhoppen commented:
nukeops should not have these

Iamgoofball commented:
Nuke ops are gonna get wrasslin instead

Zilenan91 commented:
Yeah nuke ops shouldn't have these.

PKPenguin321 commented:
>put up forum poll to get feedback that i will react to in a day or two
>a day later the forum dies

welp sit tight folks we're gonna have to wait on that to come back before this PR will progress anywhere

Iamgoofball commented:
I was actually reading this thread when forums ded, it was 14(?) votes for add to all of sec leading.

PKPenguin321 commented:
okay new plan, i'm remodeling this PR into JUST the code upgrades and leaving the addition to sec as a separate PR

PKPenguin321 committed:
fixes conflicts, removes sec implementation from security.dm

phil235 commented:
You can make the gloves temperature proof by just adding
```
 	cold_protection = HANDS
 	min_cold_protection_temperature = GLOVES_MIN_TEMP_PROTECT
 	heat_protection = HANDS
 	max_heat_protection_temperature = GLOVES_MAX_TEMP_PROTECT
 	burn_state = FIRE_PROOF
```

no need to change the kravmaga gloves's typepath to gloves/color/black.

phil235 commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -45,6 +46,9 @@
 /obj/item/clothing/gloves/color/black/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
 	if(istype(W, /obj/item/weapon/wirecutters))
+		if(!can_be_cut)
+			..()
```
the `..()` isn't needed. If you really want the parent to be called in all cases, just do if(can_be_cut) stuff stuff

PKPenguin321 committed:
does what phil says

PKPenguin321 commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -45,11 +46,12 @@
-			new /obj/item/clothing/gloves/fingerless(user.loc)
-			qdel(src)
+		if(can_be_cut)
+			if(icon_state == initial(icon_state)) //only if not dyed
```
can this and the line above it be made into one if( && ) check

phil235 replied:
if you want :shrug:

PKPenguin321 committed:
glove madness round 2

PKPenguin321 committed:
removes a comment by remie's request
this is the worst commit ive ever made
